tests/test_core.py

```diff
@@ -341,7 +350,13 @@ def test_init(self):
 
         #test invalid value initialization
         with self.assertRaises(ValueError):
-            v = relentless.core.Variable(value='4')
+            v.value = '4'
+        #test invalid low initialization
+        with self.assertRaises(ValueError):
+            v.low = '4'
+        #test invalid high initialization
+        with self.assertRaises(ValueError):
+            v.high = '4'
```

Comment:
Can you add a test or two checking that you can set low and high to real values, and that the clamped values are updated appropriately? This should probably a separate test called `test_bounds`.